src/Middleware/InitStateMiddleware.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace TractorCow\Fluent\Middleware;
+
+use SilverStripe\Admin\AdminRootController;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Middleware\HTTPMiddleware;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injector;
+use TractorCow\Fluent\Extension\FluentDirectorExtension;
+use TractorCow\Fluent\Model\Domain;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * InitStateMiddleware initialises the FluentState object and sets the current request locale and domain to it
+ */
+class InitStateMiddleware implements HTTPMiddleware
+{
+    use Configurable;
+
+    /**
+     * @var HTTPRequest
+     */
+    protected $request;
+
+    /**
+     * URL paths that should be considered as admin only, i.e. not frontend
+     *
+     * @config
+     * @var array
+     */
+    private static $admin_url_paths = [
+        'dev/',
```

Comment:
It's added in the method that uses it since it's dynamic. I could add it as a string here anyway and filter duplicates?